### PR TITLE
Improve modeling of `TokenSpec` in CodeGeneration

### PR DESCRIPTION
**Pull request by Matejkob**
Resolves part of #1977

### CODE REVIEW

File: CodeGeneration/Sources/SyntaxSupport/TokenSpec.swift [hunk 1]
```diff
@@ -21,16 +21,48 @@ public struct TokenSpec {
...
+  /// The kind of the token.
   public let kind: Kind
 
+  /// The text of the token enclosed in double quotes, if available.
+  ///
+  /// This property is used in code generation where a string representation of the token is needed.
+  public var quotedText: TokenSyntax? {
+    guard let text = text else { return nil }
+    return "\"\(raw: text)\""
+  }
```
Comment (ahoppen) [inline CodeGeneration/Sources/SyntaxSupport/TokenSpec.swift:48]: Same comment as in https://github.com/apple/swift-syntax/pull/2076#discussion_r1296182022

**Review by ahoppen — COMMENTED** [review]
One comment, otherwise LGTM. And thanks for writing those doc comments as well.

File: CodeGeneration/Sources/SyntaxSupport/TokenSpec.swift (hunk 1 above)
Comment (Matejkob) [reply CodeGeneration/Sources/SyntaxSupport/TokenSpec.swift]: Great point. Done

**Review by ahoppen — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test

**Review by kimdv — APPROVED** [review]

**Comment by Matejkob** [comment]
Somehow I've missed one usage of `raw: text`. I've just pushed fix

**Comment by kimdv** [comment]
@swift-ci please test

**Comment by kimdv** [comment]
@swift-ci please test windows

**Pull request merged**